parse.go

```diff
@@ -24,48 +22,71 @@ type Section interface {
 func Parse(v string) (GppContainer, error) {
 	var gpp GppContainer
 
-	gpp.SectionStrings = strings.Split(v, "~")
+	sectionStrings := strings.Split(v, "~")
 
-	buff := []byte(gpp.SectionStrings[0])
-	decoded := make([]byte, base64.RawURLEncoding.DecodedLen(len(buff)))
-	n, err := base64.RawURLEncoding.Decode(decoded, buff)
+	bs, err := util.NewBitStreamFromBase64(sectionStrings[0])
 	if err != nil {
 		return gpp, err
 	}
-	decoded = decoded[:n:n]
-
-	bs := util.NewBitStream(decoded)
+	if bs.Len() < 3 {
+		return gpp, fmt.Errorf("GPP Parse: a GPP string should be at least 3 bytes long")
+	}
 
-	gppType, err := util.ReadByte6(bs)
 	if err != nil {
 		return gpp, err
 	}
-	if gppType != 3 {
-		return gpp, fmt.Errorf("GPP Parse: a GPP string header must have type=3, got %d", gppType)
+	if sectionStrings[0][0] != constants.SectionGPPByte {
+		return gpp, fmt.Errorf("GPP Parse: a GPP string header must have type=%d", constants.SectionGPP)
 	}
+	bs.SetPosition(6)
```

Comment:
Can the `SetPosition` either use a constant or maybe just a comment to explain the position change?

Reply:
Not a bad idea to add a comment.